.github/workflows/qemu-jobs.yml

```diff
@@ -0,0 +1,178 @@
+name: Build and check tests for QFlex enabled and base QEMU
+
+on: 
+  push:
+    branches:
+      - master
+      - dev
+  pull_request:
+    branches:
+      - master
+      - dev
+  workflow_dispatch:
+
+jobs:
+  build_and_check_job_definition: # native_build_job_definition:
+    runs-on: ubuntu-latest
+    name: Configure, build and check QEMU # native_build_job_definition
+    strategy:
+      matrix:
+        include:
+          - # NOTE: --enable-fdt=system --enable-slirp=system was removed
+            build-name: Build for Ubuntu 20.04 + check # build-system-ubuntu
+            targets: aarch64-softmmu
+            make-check-args: check
+          - build-name: Build for user-mode + check-tcg # build-user
+            targets: aarch64-linux-user
+            configure-args: --disable-tools --disable-system
+            make-check-args: check-tcg
+          - build-name: Build for system-mode + check-tcg # build-some-softmmu
+            targets: aarch64-softmmu
+            configure-args: --disable-tools --enable-debug
+            make-check-args: check-tcg
+#          - build-name: build-user-plugins
+#            targets: aarch64-linux-user
+#            configure-args: --disable-tools --disable-system --enable-plugins --enable-debug-tcg
+#            make-check-args: check-tcg
+          - build-name: Build for system-mode with plugins enabled + check-tcg # build-some-softmmu-plugins
+            targets: aarch64-softmmu
+            configure-args: --disable-tools --disable-user --enable-plugins --enable-debug-tcg
+            make-check-args: check-tcg
+          - build-name: Build using clang + check-tcg, check-qtest # clang-system
+            targets: aarch64-softmmu
+            configure-args: --cc=clang --cxx=clang++
+              --extra-cflags=-fsanitize=undefined --extra-cflags=-fno-sanitize-recover=undefined
+            make-check-args: check-qtest check-tcg
```

Comment:
If we are going to do a test sweep with ASAN, why would we only want to run  `qtest` and `tcg`? Shouldn't we run the full suite?

Reply:
In their CI, they didn't run the full suite but `qtest` and `tcg` in system-mode with clang, and `unit` and `tcg` in user-mode with clang.

The first choice that comes to mind in this context is:
1. Do as you proposed and run the full suite in system-mode
2. Do as they did, and add also clang user-mode test for the `check-unit`, assuming that the reason they only run these tests is because rest of the behaviour was tested in other jobs and that there's a good reason to run `check-unit` in user-mode instead of system-mode